src/config.ts

```diff
@@ -70,12 +76,16 @@ export interface BundleOptions {
 export interface SnowpackPlugin {
   /** name of the plugin */
   name: string;
-  /** file extensions this plugin takes as input (e.g. [".jsx", ".js", …]) */
-  input: string[];
-  /** file extensions this plugin output (e.g. [".js", ".css"]) */
-  output: string[];
-  /** transform input to output */
-  build?(options: BuildOptions): Promise<BuildResult | null>;
+  resolve?: {
+    /** file extensions that this load function takes as input (e.g. [".jsx", ".js", …]) */
+    input: string[];
+    /** file extensions that this load function outputs (e.g. [".js", ".css"]) */
+    output: string[];
```

Comment:
One note: I think we’ve said `output` may be optional for plugin authors. Either that should be reflected here, or we should make it optional in the exposed type.

I’ve been thinking in the future maybe we have a stricter type we use internally because null-checking sucks, and we expose a looser type to users because we normalize everything on load.

WDYT?

Reply:
I guess what’s worse: two types? Or null checks everywhere?

Reply:
talked in person: now that we have a transform function for transformations that don't change file extension, load is really only for loading a source file and converting it to a web language (JS, CSS, etc.).  In that case, there's no use-case for loading a file and not changing its file extension.